### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by hoenirvili**
Refactored the code to clean up the api madness of err != nil, also there is authority var that I couldn't figure out where to place it.

### CODE REVIEW

File: cmd/cmd.go [hunk 1]
```diff
@@ -15,31 +15,45 @@ type Command struct {
...
+
 	// var authority string
 	if server == "" {
 		server = "."
 	}
+
+	// TODO(gabi) where authorithy should go?
 	if domain != "" {
-		authority = fmt.Sprintf("Kerberos:%s", domain)
+		//authority = fmt.Sprintf("Kerberos:%s", domain)
```
Comment (gabriel-samfira) [inline cmd/cmd.go]: This should go as a parameter to NewConnection. See relevant documentation

**Comment by hoenirvili** [comment]
@gabriel-samfira I'm trying right now to run the code `example/` you provided in the repo  but I'm not entirly sure why they don't currently work. This code was tested on a  **Windows 2012 R2** inside a virtual machine. 
The `command.go` hangs, also the `vmswitch.go`, (code that's creating a virtual switch) it's printing the following output. 

```
Creating br 100
Setting external port to: Intel(R) PRO/1000 NT Network Connection #2
Removing ports from
deleting
Error: Exception occured.(Invalid namespace )
```

It seems that **br 100** it's not setting up. Digging further I've added a _TODO_ in the `/virtual/network.go` file on the place where the **Error:** is coming from.

Could you please @gabriel-samfira test the code example/ on you machine and try to explain to me why I'm getting this error?

**Comment by gabriel-samfira** [comment]
You need the hyper-v feature installed to have that namespace available. Alternatively you can use the free hyper-v server edition.

**Comment by hoenirvili** [comment]
@gabriel-samfira you verified that the code is working?

**Comment by gabriel-samfira** [comment]
On my test vm, yes. Running Windows Hyper-V server 2012 R2

**Comment by hoenirvili** [comment]
@gabriel-samfira so merge?

**Comment by hoenirvili** [comment]
Do you have some time to fully review this PR?

**Comment by hoenirvili** [comment]
So I think I will close this for know, because the PR is to old.

**Pull request closed without merging**